pkg/ring/ring.go

```diff
@@ -195,47 +212,61 @@ func (r *Ring) getInternal(key uint32, n int, op Operation) ([]*IngesterDesc, er
 		distinctHosts[token.Ingester] = struct{}{}
 		ingester := r.ringDesc.Ingesters[token.Ingester]
 
-		// Ingesters that are not ACTIVE do not count to the replication limit. We do
-		// not want to Write to them because they are about to go away, but we do
-		// want to write the extra replica somewhere.  So we increase the size of the
-		// set of replicas for the key.  This means we have to also increase the
+		// We do not want to Write to Ingesters that are not ACTIVE, because they are
+		// about to go away, but we do want to write the extra replica somewhere.
```

Comment:
Nit: could be pending or joining as well as leaving, so "about to go away" is not a full explanation.